src/components/main-view/main-view.jsx

```diff
@@ -1,38 +1,133 @@
+  const storedUser = localStorage.getItem('user');
+  const storedToken = localStorage.getItem('token');
+
+
+
+
 
   useEffect(() => {
-    fetch("https://myflix-api-3dxz.onrender.com/movies")
+    if (!token) {
+      console.log("No token")
+      return;
+    }
+
+    if (user && token) {
+      console.log(user ,token)
+      fetch("https://myflix-api-3dxz.onrender.com/movies", {
+      headers: { Authorization: `Bearer ${token}` },
+      })
       .then((response) => response.json())
-      .then((data) => {
-        const moviesFromApi = data.docs.map((doc) => {
-          return {
-            id: doc.key,
-            title: doc.title,
-            author: doc.director_name?.[0]
-          };
-        });
-
-        setMovies(moviesFromApi);
+      .then((movies) => {
+        setMovies(movies);
+        
       });
+    }
   }, []);
 
+  const handleMovieClick = (movie) => {
+    setSelectedMovie(movie);
+    console.log("State selectedMovie", selectedMovie)
+  }
+
+  const onLoggedIn = (user, token) => {
+    setUser(user);
+    setToken(token);
+  }
+
   if (selectedMovie) {
     return (
-      <MovieView movie={selectedMovie} onBackClick={() => setSelectedMovie(null)} />
+      <>
+        <button
+          onClick={() => {
+            setUser(null); setToken(null); localStorage.clear();
+          }}
+        >
+          Logout
+        </button>
+        <MovieView
+          movie={selectedMovie}
+          onBackClick={() => setSelectedMovie(null)}
+        />
+      </>
     );
- }
+  }
+    
+  if (user && token) {
+    if (movies.length === 0) {
+      
+      return <div> The list is empty! </div>;
+    }
+    return <div>
+      <button onClick={() => {setUser(null); setToken(null); localStorage.clear();}}>
+        Logout
+      </button>
+  {movies.map((movie) => <MovieCard movie={movie} onMovieClick={handleMovieClick}></MovieCard>)}</div>;
+  } 
+  else {
+    return <>
+    <LoginView onLoggedIn={onLoggedIn}/>
+    <SignupView />
+    </>;
+  }
+}
+
+/*
 
   if (movies.length === 0) {
-    return <div> The list is empty! </div>;
+    return (
+      <>
+        <button
+          onClick={() => {
+            setUser(null); setToken(null); localStorage.clear();
+          }}
+        >
+          Logout
+        </button>
+        <div>The list is empty!</div>
+      </>
+    );
   }
 
+  return (
+    <div>
+      <button
+        onClick={() => {
+          setUser(null); setToken(null); localStorage.clear();
+        }}
+      >
+        Logout
+      </button>
+      {movies.map((movie) => (
+        <MovieCard
+          key={movie.id}
+          movie={movie}
+          onMovieClick={(newSelectedMovie) => {
+            setSelectedMovie(newSelectedMovie);
+          }}
+        />
+      ))}
+    </div>
+  );
+  
+
+/*  if (selectedMovie) {
+    return (
+      <MovieView movie={selectedMovie} onBackClick={() => setSelectedMovie(null)} />
+    );
+ }
+
```

Comment:
Commented out code clutters the code base, let us not commit them. You can always checkout to a previous commit if you want to see old code.